includes/class-amp-theme-support.php

```diff
@@ -1966,18 +1966,25 @@ public static function prepare_response( $response, $args = [] ) {
 			header( 'Content-Type: text/html; charset=utf-8' );
 		}
 
-		// @todo Both allow_dirty_styles and allow_dirty_scripts should eventually use AMP dev mode instead.
 		$args = array_merge(
 			[
 				'content_max_width'    => ! empty( $content_width ) ? $content_width : AMP_Post_Template::CONTENT_MAX_WIDTH, // Back-compat.
 				'use_document_element' => true,
-				'allow_dirty_styles'   => self::is_customize_preview_iframe(), // Dirty styles only needed when editing (e.g. for edit shortcuts).
-				'allow_dirty_scripts'  => is_customize_preview(), // Scripts are always needed to inject changeset UUID.
 				'user_can_validate'    => AMP_Validation_Manager::has_cap(),
 			],
 			$args
 		);
 
+		if ( is_customize_preview() ) {
+			// Scripts are always needed to inject changeset UUID.
+			$args['element_xpaths'][] = '//script';
+		}
+
+		if ( self::is_customize_preview_iframe() ) {
+			// Prevent the Customizer preview styles from being sanitized.
+			$args['element_xpaths'][] = '//link[ @id="customize-preview-css" ]';
```

Comment:
There could be other stylesheets that need targeting than just `customize-preview`. For example, whenever I wrote a plugin for adding CSS to the Customizer preview I'd enqueue the stylesheet with `customize-preview` as a dependency:

* https://github.com/xwp/wp-customize-featured-content-demo/blob/921ab371edc5cdfb718c317a835355cfc5cca821/php/class-plugin.php#L251-L254
* https://github.com/xwp/wp-customize-partial-refresh/blob/b95283a9d8ec0288a8bf9ab7d4a153984d4df4a5/php/class-wp-customize-selective-refresh.php#L145-L148
* https://github.com/xwp/wp-customize-snapshots/blob/8896256e34d9d57fefc61fec2407998eb187bc41/php/class-plugin.php#L113-L116

Such dependencies would be accounted for by `\AMP_Theme_Support::has_dependency()`.

Also note that Dashicons is a dependency of `customize-preview`. Is that properly getting included in Dev Mode This should be accounted for by `\AMP_Theme_Support::is_exclusively_dependent()`.

So I suggest rather:

```diff
diff --git a/includes/class-amp-theme-support.php b/includes/class-amp-theme-support.php
index 44c59d4e1..811b63eb8 100644
--- a/includes/class-amp-theme-support.php
+++ b/includes/class-amp-theme-support.php
@@ -1071,6 +1071,9 @@ class AMP_Theme_Support {
 		add_action( 'wp_enqueue_scripts', [ __CLASS__, 'enqueue_assets' ], 0 ); // Enqueue before theme's styles.
 		add_action( 'wp_enqueue_scripts', [ __CLASS__, 'dequeue_customize_preview_scripts' ], 1000 );
 		add_filter( 'customize_partial_render', [ __CLASS__, 'filter_customize_partial_render' ] );
+		if ( is_customize_preview() ) {
+			add_filter( 'style_loader_tag', [ __CLASS__, 'filter_admin_bar_style_loader_tag' ], 10, 2 );
+		}
 
 		add_action( 'wp_footer', 'amp_print_analytics' );
 
@@ -1559,6 +1562,28 @@ class AMP_Theme_Support {
 		return $tag;
 	}
 
+	/**
+	 * Add data-ampdevmode attribute to any enqueued style in the customizer preview.
+	 *
+	 * @since 1.6
+	 *
+	 * @param string $tag    The link tag for the enqueued style.
+	 * @param string $handle The style's registered handle.
+	 * @return string Tag.
+	 */
+	public static function filter_customize_preview_style_loader_tag( $tag, $handle ) {
+		$customize_preview = 'customize-preview';
+		if (
+			is_array( wp_styles()->registered[ $customize_preview ]->deps ) && in_array( $handle, wp_styles()->registered[ $customize_preview ]->deps, true ) ?
+				self::is_exclusively_dependent( wp_styles(), $handle, $customize_preview ) :
+				self::has_dependency( wp_styles(), $handle, $customize_preview )
+		) {
+			$tag = preg_replace( '/(?<=<link)(?=\s|>)/i', ' ' . AMP_Rule_Spec::DEV_MODE_ATTRIBUTE, $tag );
+		}
+
+		return $tag;
+	}
+
 	/**
 	 * Add data-ampdevmode attribute to any enqueued script that depends on the admin-bar.
 	 *
```

Reply:
Yea good point. Addressed in https://github.com/ampproject/amp-wp/pull/4977/commits/927ed5fd0102233fb2343993b8aa9e33ad970305.